### PR TITLE
Suggestion Navigation Buttons

### PR DESCRIPTION
**Pull request by bhagyakjain**
- Now, only one suggestion is visible at a time
- The user can navigate the suggestions. 
- First suggestion in the list will be displayed first

### CODE REVIEW

File: ui/ui_vfinal/intents/javascript_functions/submit_query.html [hunk 1]
```diff
@@ -208,6 +174,109 @@
...
+      index++;
+      if(index >= suggestions.length - 1) {
+          // hide next button
+          hideNextButton();
+      }
+      else {
+          displayNextButton();
+      }
+    
+      if(index >= suggestions.length)
```
Comment (adimehta) [inline ui/ui_vfinal/intents/javascript_functions/submit_query.html:207]: Let's follow the consistent style throughout the code : 
if (cond) {
  // Some code
} else {
  // some other code
}

File: ui/ui_vfinal/intents/javascript_functions/submit_query.html [hunk 2]
```diff
@@ -219,11 +288,19 @@
...
     if (suggestionsContent.style.maxHeight){
       suggestionsContent.style.maxHeight = null;
+      hideNextButton();
+      hidePreviousButton();
+      index=0;
     } 
     else {
       suggestionsContent.style.maxHeight = suggestionsContent.scrollHeight + 'px';
+      // index check rqd
+      index=0;
```
Comment (adimehta) [inline ui/ui_vfinal/intents/javascript_functions/submit_query.html:298]: We are resetting the index to 0, but not calling showSuggestions(index) again to show the correct suggestion.

File: ui/ui_vfinal/intents/javascript_functions/submit_query.html [hunk 3]
```diff
@@ -152,45 +152,11 @@
...
-      }
-      messages += '</div>';
-      
-      messages += '</div>';
-    }
-    document.querySelector('#suggestions-content').innerHTML = messages;
+    index = 0;
+    showSuggestion(index);
+    
+    hideNextButton();
```
Comment (adimehta) [inline ui/ui_vfinal/intents/javascript_functions/submit_query.html:158]: If there are more suggestions, shouldn't this be displayNextButton then?

Comment (bhagyakjain) [reply ui/ui_vfinal/intents/javascript_functions/submit_query.html:158]: removing 
`
index = 0;
showSuggestion(index);
`
when the page loads initially, we should hide the buttons as the suggestion dropdown is closed. But when it opens it resets the index to zero.
So, these 2 lines are are redundant.

File: ui/ui_vfinal/intents/javascript_functions/submit_query.html (hunk 1 above)
Comment (bhagyakjain) [reply ui/ui_vfinal/intents/javascript_functions/submit_query.html:204]: Done.

File: ui/ui_vfinal/intents/javascript_functions/submit_query.html (hunk 2 above)
Comment (bhagyakjain) [reply ui/ui_vfinal/intents/javascript_functions/submit_query.html:294]: Corrected. we only will only call `showSuggestions(index)` only when the suggestion list is not empty.

File: ui/ui_vfinal/intents/javascript_functions/submit_query.html (hunk 1 above)
Comment (adimehta) [reply ui/ui_vfinal/intents/javascript_functions/submit_query.html:204]: Let's do it for this if also. 
if (index >= suggestions.length) {
  return;
}

**Pull request merged**